src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -435,40 +486,47 @@ class SearchBarUI extends Component<SearchBarProps, State> {
       );
     }
 
-    let filterBar;
-    if (this.shouldRenderFilterBar()) {
-      const filterGroupClasses = classNames('globalFilterGroup__wrapper', {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        'globalFilterGroup__wrapper-isVisible': this.state.isFiltersVisible,
-      });
-      filterBar = (
-        <div
-          id="GlobalFilterGroup"
-          ref={(node) => {
-            this.filterBarWrapperRef = node;
-          }}
-          className={filterGroupClasses}
-        >
-          <div
-            ref={(node) => {
-              this.filterBarRef = node;
-            }}
-          >
-            <FilterBar
-              className="globalFilterGroup__filterBar"
-              filters={this.props.filters!}
-              onFiltersUpdated={this.props.onFiltersUpdated}
-              indexPatterns={this.props.indexPatterns!}
-            />
-          </div>
-        </div>
+    let queryEditor;
+    if (this.shouldRenderQueryEditor(isEnhancementsEnabledOverride)) {
+      queryEditor = (
+        <QueryEditorTopRow
+          timeHistory={this.props.timeHistory}
+          containerRef={this.props.containerRef}
+          settings={this.props.settings}
+          query={this.state.query}
+          screenTitle={this.props.screenTitle}
+          onSubmit={this.onQueryBarSubmit}
+          indexPatterns={this.props.indexPatterns}
+          isLoading={this.props.isLoading}
+          prepend={this.props.showFilterBar ? savedQueryManagement : undefined}
+          showDatePicker={this.props.showDatePicker}
+          dateRangeFrom={this.state.dateRangeFrom}
+          dateRangeTo={this.state.dateRangeTo}
+          isRefreshPaused={this.props.isRefreshPaused}
+          refreshInterval={this.props.refreshInterval}
+          showAutoRefreshOnly={this.props.showAutoRefreshOnly}
+          showQueryEditor={this.props.showQueryInput}
+          onRefresh={this.props.onRefresh}
+          onRefreshChange={this.props.onRefreshChange}
+          onChange={this.onQueryBarChange}
+          isDirty={this.isDirty()}
+          customSubmitButton={
+            this.props.customSubmitButton ? this.props.customSubmitButton : undefined
```

Comment:
nit: `this.props.customSubmitButton || undefined`

Reply:
+1,

added here when we do clean up of both components: